NBitcoin/TransactionBuilder.cs

```diff
@@ -1623,7 +1639,6 @@ public Transaction BuildTransaction(bool sign, SigHash sigHash)
 		public Transaction BuildTransaction(bool sign, SigningOptions signingOptions)
 		{
 			int totalRepass = 5;
-			DoShuffleGroups();

```

Comment:
groups are shuffled before the rest, there is a reason for that, probably why tests break.

Reply:
But in my PR DoShuffleGroups() is the first thing that DoShuffle() does.

Reply:
No your code is not equivalent, you are not shuffling group before building the transaction, which make it useless.

Reply:
ah, so with "the rest" you don't mean "shuffling the rest" but more stuff, got it; sorry for taking your time